### PR TITLE
Add support for billing thresholds

### PR DESCRIPTION
**Pull request by remi-stripe**
r? @brandur-stripe 
cc @stripe/api-libraries 

This one is a bit complex as it's across multiple resources and no specific test was added for it so we need to make sure it matches OpenAPI.

cc @clintonb-stripe Could you double check that I added the right set of properties too and I'm not missing any? (Java only lists properties, not parameters).

Docs: https://stripe.com/docs/billing/subscriptions/metered-billing/thresholds
Internal PR with gate removal: #126799

### CODE REVIEW

**Comment by remi-stripe** [comment]
(like the other one, removing the approval to leave that to Brandur. Thanks for the review Clinton!)

File: invoice.go [hunk 1]
```diff
@@ -217,6 +219,19 @@ type InvoiceCustomField struct {
 	Value string `json:"value"`
 }
 
+// InvoiceThresholdReason is a structure representing a reason for a billing threshold.
+type InvoiceThresholdReason struct {
+	AmountGte   int64                               `json:"amount_gte"`
```
Comment (brandur-stripe) [inline invoice.go]: I'm assuming gte = "greater than or equal". Can we upcase it in all instances to be `GTE`? Things like `ID`, `PDF`, or `URL` are pretty ubiquitous convention through the library.

File: invoice.go [hunk 2]
```diff
@@ -217,6 +219,19 @@ type InvoiceCustomField struct {
...
+	ItemReasons []*InvoiceThresholdReasonItemReason `json:"item_reasons"`
+}
+
+// InvoiceThresholdReasonItemReason is a structure representing the line items that
+// triggered an invoice.
+type InvoiceThresholdReasonItemReason struct {
+	LineItemIds []string `json:"line_item_ids"`
```
Comment (brandur-stripe) [inline invoice.go]: And similarly here: `IDs`.

**Comment by remi-stripe** [comment]
@brandur-stripe Fixed, PTAL (and thank you)

**Comment by brandur-stripe** [comment]
LGTM!

**Comment by brandur-stripe** [comment]
Released as 55.11.0.

**Pull request merged**